src/WorldStorage/FastNBT.cpp

```diff
@@ -30,85 +30,78 @@ static const int MAX_LIST_ITEMS = 10000;
 	#define PROPAGATE_ERROR(X) do { auto Err = (X); if (Err != eNBTParseError::npSuccess) return Err; } while (false)
 #endif
 
+#define AS_INT(x) static_cast<int>(x)
+
 
 
 ////////////////////////////////////////////////////////////////////////////////
 // cNBTParseErrorCategory:
 
 AString cNBTParseErrorCategory::message(int a_Condition) const
 {
-	switch (static_cast<eNBTParseError>(a_Condition))
+	switch (a_Condition)
```

Comment:
Why? Your macro should be the fix for the clang warning (as you've done it in the other 90%). Not some dirty `AS_INT` trick (without even explaining, why it is necessary).
Then I also don't need the `AS_INT` macro.

Reply:
Because the switch isn't covered.  The input is an int and therefore could take any value.

I rewrote it to use `int` in the switch to avoid the default being optimised away as comments elsewhere suggest.  `AS_INT` is simply less visual noise than `static_cast<int>()`.

Reply:
It would be better to change `eNBTParseError` from `enum class` to a simple `enum` with a namespace around it. Then put a comment above, which explains, that `std::error_category` requires this to be an integer and therefore we don't need strict type-checking.
Now you can remove all the `static_cast` from the `eNBTParseError` enum.
As a bonus, you could add a typedef: `using eNBTParseError_t = int`

Reply:
I'm not keen on the idea of reducing the type safety globally just to save some typing within a single function.

>`using eNBTParseError_t = int`

This wouldn't allow assigning to a `std::error_code`: I can't overload `std::is_error_code_enum` for `int`...

Reply:
> error_condition( int val, const error_category& cat );
> template< class ErrorConditionEnum > error_condition( ErrorConditionEnum e );

Sorry, I now see, that `std::error_condition` is the problem. Allowing things like `std::error_condition(77, std::iostream_category())` just kills the type safety.

The `value()` function just shows, that this was designed for C-Enums:
```cpp
if( errorCode.category() == boost::asio::error::misc_category )
    {
        switch (errorCode.value())
        {
        case boost::asio::error::eof:
            debugMsg << ". Server has disconnected.";
            break;
        case boost::asio::error::connection_refused:
            debugMsg << ". Connection Refused";
            break;
        default:
            debugMsg << ". Unknown Error.";
            break;
        }
    }
```
And then somebody came ahead and just glued the Static typed enums at this class with `is_error_condition_enum` and `template< class ErrorConditionEnum > error_condition( ErrorConditionEnum e );`. And this lead us to subtile errors like the above.

I'm ok with your way. But I would suggesst using a template instead of a macro (so I can do `as_int(eNBTParseError::npNeedBytes)`, but not `as_int(cEntityEffect::effWither)`). This also removes the `#undef`.